src/main/perl/Check.pm

```diff
@@ -0,0 +1,321 @@
+${PMpre} CAF::Check${PMpost}
+
+use CAF::Object;
+use LC::Check;
+use LC::Exception qw (SUCCESS throw_error);
+use Readonly;
+
+use File::Path qw(rmtree);
+use File::Copy qw(move);
+
+Readonly my $KEEPS_STATE => 'keeps_state';
+
+Readonly::Hash my %CLEANUP_DISPATCH => {
+    move => \&move,
+    rmtree => \&rmtree,
+    unlink => sub { return unlink(shift); },
+};
+
+# Use dispatch table instead of non-strict function by variable call
+Readonly::Hash my %LC_CHECK_DISPATCH => {
+    directory => \&LC::Check::directory,
+};
+
+our $EC = LC::Exception::Context->new->will_store_all;
+
+=pod
+
+=head1 NAME
+
+CAF::Check - check that things are really the way we expect them to be
+
+=head1 DESCRIPTION
+
+Simplify common file and directory related operations e.g.
+
+=over
+
+=item directory creation
+
+=item cleanup
+
+=item (mockable) file/directory tests
+
+=back
+
+The class is based on L<LC::Check> with following major difference
+
+=over
+
+=item C<CAF::Object::NoAction> support builtin (and C<keeps_state> option to override it).
+
+=item support C<CAF::Reporter> (incl. C<CAF::History>)
+
+=item raised exceptions are catched, methods return SUCCESS on succes,
+undef on failure and store the error message in the C<fail> attribute.
+
+=item available as class-methods
+
+=back
+
+=head2 Methods
+
+=over
+
+=cut
+
+# TODO: do we need some magic to be able to use as regular exported functions
+# TODO: handle LC::Check _message, should use Reporter instead of print
+
+=item _get_noaction
+
+Return NoAction setting:
+
+=over
+
+=item Return 0 is C<keeps_state> is true
+
+Any other value of C<keeps_state> is ignored. (In particular,
+you cannot use C<keeps_state> to enable NoAction).
+
+=item Return value of C<CAF::Object::NoAction> otherwise.
+
+=back
+
+Supports an optional C<msg> that is prefixed to reporter.
+
+=cut
+
+
+sub _get_noaction
+{
+    my ($self, $keeps_state, $msg) = @_;
+
+    $msg = '' if (! defined($msg));
+
+    my $noaction;
+
+    if ($keeps_state) {
+        $self->verbose($msg, "keeps_state set, noaction is false");
```

Comment:
verbose seems too high here for something that's not informative to a user. The user needs to see the normal log message modified to confirm that no action was actually taken, this sort of trace level message is only useful to developers and should go into debug()